breacharbiter_test.go

```diff
@@ -831,3 +925,596 @@ restartCheck:
+	if err != nil {
+		t.Fatalf("unable to determine if channel is "+
+			"breached: %v", err)
+	}
+
+	if !isBreached {
+		t.Fatalf("channel %v was never marked breached",
+			chanPoint)
+	}
+
+}
+
+// assertNoArbiterBreach checks that the breach arbiter has not persisted the
+// breach information for a particular channel.
+func assertNoArbiterBreach(t *testing.T, brar *breachArbiter,
+	chanPoint *wire.OutPoint) {
+
+	isBreached, err := brar.IsBreached(chanPoint)
+	if err != nil {
+		t.Fatalf("unable to determine if channel is "+
+			"breached: %v", err)
+	}
+
+	if isBreached {
+		t.Fatalf("channel %v was marked breached",
+			chanPoint)
+	}
+}
+
+// assertPendingClosed checks that the channel has been marked pending closed in
+// the channel database.
+func assertPendingClosed(t *testing.T, c *lnwallet.LightningChannel) {
+	closedChans, err := c.State().Db.FetchClosedChannels(true)
+	if err != nil {
+		t.Fatalf("unable to load pending closed channels: %v", err)
+	}
+
+	for _, chanSummary := range closedChans {
+		if chanSummary.ChanPoint == *c.ChanPoint {
+			return
+		}
+	}
+
+	t.Fatalf("channel %v was not marked pending closed",
+		c.ChanPoint)
+}
+
+// assertNotPendingClosed checks that the channel has not been marked pending
+// closed in the channel database.
+func assertNotPendingClosed(t *testing.T, c *lnwallet.LightningChannel) {
+	closedChans, err := c.State().Db.FetchClosedChannels(true)
+	if err != nil {
+		t.Fatalf("unable to load pending closed channels: %v", err)
+	}
+
+	for _, chanSummary := range closedChans {
+		if chanSummary.ChanPoint == *c.ChanPoint {
+			t.Fatalf("channel %v was marked pending closed",
+				c.ChanPoint)
+		}
+	}
+}
+
+// createTestArbiter instantiates a breach arbiter with a failing retribution
+// store, so that controlled failures can be tested.
+func createTestArbiter(t *testing.T, notifier chainntnfs.ChainNotifier,
+	db *channeldb.DB) (*breachArbiter, func(), error) {
+
+	// Create a failing retribution store, that wraps a normal one.
+	store := newFailingRetributionStore(func() RetributionStore {
+		return newRetributionStore(db)
+	})
+
+	aliceKeyPriv, _ := btcec.PrivKeyFromBytes(btcec.S256(),
+		alicesPrivKey)
+	signer := &mockSigner{key: aliceKeyPriv}
+
+	// Assemble our test arbiter.
+	ba := newBreachArbiter(&BreachConfig{
+		CloseLink:          func(_ *wire.OutPoint, _ htlcswitch.ChannelCloseType) {},
+		DB:                 db,
+		Estimator:          &lnwallet.StaticFeeEstimator{FeeRate: 50},
+		GenSweepScript:     func() ([]byte, error) { return nil, nil },
+		Notifier:           notifier,
+		Signer:             signer,
+		PublishTransaction: func(_ *wire.MsgTx) error { return nil },
+		Store:              store,
+	})
+
+	if err := ba.Start(); err != nil {
+		return nil, nil, err
+	}
+
+	// The caller is responsible for closing the database.
+	cleanUp := func() {
+		ba.Stop()
+	}
+
+	return ba, cleanUp, nil
+}
+
+// createInitChannelsWithNotifier creates two initialized test channels funded
```

Comment:
It's out of the scope of this PR, but we should really extract this function into a package as a public test helper function. This instances counts the 4th location within the codebase this function has been copied.

For example, there's already a `createTestPeer` in `test_utils.go` that can be re-used here.